packages/kit/src/cli.js

```diff
@@ -62,35 +63,57 @@ prog
 		let close;
 
 		async function start() {
-			const { dev } = await import('./core/dev/index.js');
+			const svelte_config = await load_config();
+			const { svelte, sveltekit } = await import('./core/dev/plugin.js');
+			const vite_config = await svelte_config.kit.vite();
+
+			/** @type {import('vite').UserConfig} */
+			const config = {
+				plugins: [...(vite_config.plugins || []), svelte(svelte_config), sveltekit(svelte_config)]
+			};
+			config.server = config.server || {};
```

Comment:
```suggestion
			const config = {
				plugins: [...(vite_config.plugins || []), svelte(svelte_config), sveltekit(svelte_config)],
				server: {}
			};
```

Reply:
It's a more natural way to write it, but upsets TypeScript. TypeScript see the type annotation which says `server` can be optional and that takes precedence. If we want to do it the suggested way then I also need to add ` & {server: import('vite').ServerOptions}` to the type, which seemed even uglier to me. I just noticed the way I have it written currently can be simplified though